File: java/lab5/src/commands/ShowCommand.java
package src.commands;

import src.baseabstractions.Command;
import src.basecollection.CollectionManager;
import src.baseobjects.Product;

/**
 * Command to display all elements of the collection in string representation.
 *
 * @param <K> the type of keys used in the collection, must be Comparable
 * @param <V> the type of values stored in the collection, must extend Product
 */
public class ShowCommand<K extends Comparable<K>, V extends Product> extends Command<K, V> {

    /**
     * Constructs a ShowCommand with the specified collection manager.
     *
     * @param collectionManager the collection manager to operate on
     */
    public ShowCommand(CollectionManager<K, V> collectionManager) {
        super(collectionManager);
    }

    /**
     * Executes the show command.
     * Displays all elements of the collection in string representation.
     *
     * @param args command arguments (expects exactly one argument - the command
     *             name)
     * @throws IllegalArgumentException if the number of arguments is incorrect
     */
    public void execute(String[] args) {
        if (args.length == 1) {
            getCollectionManager().show();
        } else
            throw new IllegalArgumentException("Number of arguments is wrong.");
    }

    /**
     * Returns a description of the command.
     *
     * @return "show : вывести в стандартный поток вывода все элементы коллекции в
     *         строковом представлении" - command description in Russian
     */
    public String getDescription() {
        return "show : вывести в стандартный поток вывода все элементы коллекции в строковом представлении";
    }
}

File: java/lab6/client/src/main/kotlin/commands/ShowCommand.kt
package commands

import abstractions.Command
import kotlinx.serialization.Serializable
import kotlinx.serialization.Transient
import objects.Product

@Serializable
class ShowCommand<K : Comparable<K>, V : Product> : Command<K, V>() {
    @Transient
    override val description =
        "show : вывести в стандартный поток вывода все элементы коллекции в строковом представлении"

    @Transient
    override val isShouldBeSent = true

    private var productsCollection: HashMap<K, V> = hashMapOf()

    override fun start(args: Array<String>) {
        if (args.size != 1) {
            throw IllegalArgumentException("Number of arguments is wrong.")
        }
    }

    override fun finish() {
        if (productsCollection.isEmpty()) {
            println("The collection does not contain any items.")
        } else {
            for (product in productsCollection) {
                println("${product.key}: ${product.value}")
            }
        }
    }
}